File: 633-sum-of-square-numbers/633-sum-of-square-numbers.java
class Solution {
    public boolean judgeSquareSum(int c) {
        long a = 0;
        long b = (int)Math.sqrt(c);
        
        while(a<=b){
            long result = a*a + b*b;
            if(result == c){
                return true;
            }
            if(result > c){
                b--;
            }else{
                a++;
            }
        }
        return false;
    }
}

File: 633-sum-of-square-numbers/633-sum-of-square-numbers.js
/**
 * @param {number} c
 * @return {boolean}
 */
var judgeSquareSum = function(c) {
    let a = 0;
    let b = Math.floor(Math.sqrt(c));
    
    while(a<=b){
        let result = a*a + b*b;
        if(result === c){
            return true;
        }
        if(result > c){
            b--;
        }else{
            a++;
        }
    }
    return false;
};